docs/source/workflow/generate-types.mdx

```diff
@@ -0,0 +1,205 @@
+
+Finally, we recommend adding helpful scripts to your `package.json` file to ensure your TS types are regularly generated:
+
+```json
+{
+// ...
+  "scripts": {
+    "generate": "graphql-codegen --config codegen.yml",
+    "compile": "npm run generate && tsc",
+    "postinstall": "npm run compile",
+    "start": "npm run compile && node ./dist/index.js",
+  },
+// ...
+}
+```
+
+Above, running the `npm start` command generates types based on our GraphQL schema _and_ compiles our TypeScript code. The first time you run the `graphql-codegen` command, you'll see a file full of generated types at the path you specified in your `codegen.yml` file. 
+
+## Adding types to resolvers
+
+The `typescript-resolvers` plugin creates a `Resolvers` type that you can use to add a type to your [resolver map](../data/resolvers#base-syntax), ensuring your resolvers return values match the field types specified by your schema.
+
+Import the `Resolvers` type into the file where you define your resolvers:
+
+```ts title="resolvers.ts"
+// This is the file where our generated types live
+// (specified in our `codegen.yml` file)
+import { Resolvers } from './__generated__/resolvers-types';
+```
+
+You can now add the `Resolvers` type directly to your resolver map:
+
+```ts
+export const resolvers: Resolvers = {}
+```
+
+Your resolvers can now type check that the *arguments* and *return value* for each resolver match the schema:
+
+```ts
+export const resolvers: Resolvers = {
+  Query: {
+    // TypeScript now complains about the below resolver because
+    // the data returned by this resolver doesn't match the schema type
+    // (i.e., type Query { books: [Book] })
+    books: () => {
+      return "apple";
+    },
+  },
+}
+```
+
+### Context typing for resolvers
+
+You can also configure GraphQL Code Generator to add a type for the context your resolvers share, ensuring TypeScript warns you if you attempt to use a context value that doesn't exist.
+
+To do this, you must first export the interface you pass to Apollo Server as a generic type parameter for typing your context value:
+
+```ts title="src/index.ts"
+// highlight-start
+export interface MyContext {
+  dataSources: {
+    books: Book[];
+  };
+}
+// highlight-end
+
+const server = new ApolloServer<MyContext>({
+  typeDefs,
+  resolvers,
+});
+```
+
+Remember the `contextType` from our [`codegen.yml` file above](#installing-and-configuring-dependencies)? You can pass your exported context interface to the `contextType` configuration option, like so:
+
+```yaml
+# ...
+config:
+  useIndexSignature: true
+  # Providing our context's interface ensures our
+  # context's type is set for all of our resolvers. 
+
+  # Note, this file path starts from the location of the 
+  # file where you generate types. 
+  # (i.e., `/src/__generated__/resolvers-types.ts` above) 
+  contextType: ../index#MyContext # highlight-line
```

Comment:
The `#MyContext` gets deleted in the render here :(

Reply:
Fix this by wrapping in double quotes!